include/AudioPipe.h

```diff
@@ -15,14 +17,14 @@ class AudioPipe :
 	virtual ~AudioPipe();
 	
 	//Audio input
-	virtual int RecBuffer(SWORD *buffer,DWORD size);
+	virtual DWORD RecBuffer(DWORD frameSize, const AudioBuffer::shared& audioBuffer);
```

Comment:
Let's return it instead
```suggestion
	virtual  AudioBuffer::shared RecBuffer(DWORD frameSize);
```